planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -47,74 +51,64 @@ public Itinerary generateItinerary(List<Event> events) {
         int END = events.get(0).getOpeningHours().end();
 
         Collections.sort(events, Event.SortByOrder);
-        Duration [][] travelTimeGraph = getEventsDistanceMatrix(events);
-        if (errorMessage != null) { 
-            Itinerary itinerary = new Itinerary(Arrays.asList(), errorMessage);
-            return itinerary;
+        List<ItineraryItem> items = new ArrayList<>();
+        if (optimized) {
+            items = scheduleOptimizedItinerary(events, START, END);
+        }else {
+            items = scheduleItineraryInOrder(events, START, END);
         }
-        List<ItineraryItem> items = scheduleItineraryInOrder(events, START, END, travelTimeGraph);
+        
         Itinerary itinerary = new Itinerary(items, errorMessage);
         return itinerary;
     }
 
-    // Function that creates a graph with the events as vertices and the travelling
-    // time between the events as edges.
-    // The order of this graph will be the same as the order of the events.
-    // For the MVP, real time traffic is NOT used.
-    // TODO: handle invalid addresses
-    private Duration[][] getEventsDistanceMatrix(List<Event> events) {
-        int numOfNodes = events.size();
-        Duration[][] travelTimeGraph = new Duration[numOfNodes][numOfNodes];
-
-        String GoogleApiKey = "AIzaSyDK36gDoYgOj4AlbCqh1IuaUuTlcpKF0ns";
-        String[] origins = getListOfAddresses(events);
-        String[] destinations = getListOfAddresses(events);
-        GeoApiContext context = new GeoApiContext.Builder(new GaeRequestHandler.Builder())
-                            .apiKey(GoogleApiKey)
-                            .build();
-       try {
-            DistanceMatrix distanceMatrix = DistanceMatrixApi.getDistanceMatrix(context,
-                                                         origins,
-                                                         destinations).await();
+    // Function that creates an itinerary by scheduling each event in order.
+    private List<ItineraryItem> scheduleItineraryInOrder(List<Event> events, int openningTime, int endingTime) {
+        List<ItineraryItem> items = new ArrayList<>();
+        boolean optimized = false;
```

Comment:
I was trying to improve readability hahah (so when people see this line of code they know why we pass true/false into the function). Do you think it's redundant?

Reply:
Oh! I see! I think it's not necessary! However, I just noticed that the functions scheduleOptimizedItinerary and scheduleItineraryInOrder are 90% alike. So, maybe we can restructure them to avoid code duplication?